common-lib/common/signal/signal.libsonnet

```diff
@@ -5,52 +5,151 @@ local counter = import './counter.libsonnet';
 local gauge = import './gauge.libsonnet';
 local histogram = import './histogram.libsonnet';
 local info = import './info.libsonnet';
+local raw = import './raw.libsonnet';
+local stub = import './stub.libsonnet';
 {
+  //Expected signalsJson format:
+  // {
+  //   filteringSelector: 'job!=""',
+  //   groupLabels: ['job'],
+  //   instanceLabels: ['instance'],
+  //   discoveryMetric: 'up',
+  //   aggLevel: 'group' or 'instance' or 'none',
+  //   aggFunction: 'avg',
+  //   signals: {
+  //     signal1: {
+  //       name: 'Golang version',
+  //       type: 'info',
+  //       description: 'Golang version used.',
+  //       expr: 'go_info{%(queriesSelector)s}',
+  //       infoLabel: 'version',
+  //     },
+  //     signal2:...,
+  //     signal3:....
+  //   }
+  // }
+  unmarshallJson(signalsJson):
```

Comment:
This is actually extraordinarily powerful and impactful.

This can allow us to COMPLETELY decouple the desired signals from the resulting dashboard, we could even store the signals in a separate file like `signals.json`, and allow it to be loaded by the mixin, or used as a basis for others dashboards.

When you added this, how did you envision it being used? Could you elaborate a bit on the use-cases in the README?

Reply:
Thanks. Indeed, decoupling is the key. I also wanted to keep signals as part of jsonnet config or event json, so they can be modified by users before generating final dashboards and so on. Also I keep grafana foundation sdk in mind, so perhaps one days those json files with signals can be used to generate dashboards using Go or TypeScript as well.

Reply:
For now, I think docker lib update can service as good use case example.